linter.py

```diff
@@ -1,58 +1,32 @@
+        # The default implementation will look for a user defined `executable`
+        # setting.
+        success, executable = super().context_sensitive_executable_path(cmd)
+        if success:
+            return True, executable
+
+        settings = self.get_view_settings()
+        gem_name = cmd[0]
+        if settings.get('use_bundle_exec', False):
+            return True, ['bundle', 'exec', gem_name]
+
+        return True, gem_name  # Avoid testing if `gem_name` is on PATH
 
 
 class Rubocop(RubyLinter):
-    defaults = {
-        'selector': 'source.ruby - text.html - text.haml'
-    }
+    cmd = ['rubocop', '--format', 'emacs', '--force-exclusion', '${args}']
+
     regex = (
         r'^.+?:(?P<line>\d+):(?P<col>\d+): '
         r'(:?(?P<warning>[RCW])|(?P<error>[EF])): '
         r'(?P<message>.+)'
     )
 
-    def cmd(self):
-        """Build command, using STDIN if a file path can be determined."""
-
-        settings = self.get_view_settings()
-
-        command = ['ruby', '-S']
-
-        if settings.get('use_bundle_exec', False):
-            command.extend(['bundle', 'exec'])
-
-        command.extend(['rubocop', '--format', 'emacs'])
-
-        # Set tempfile_suffix so by default a tempfile is passed onto rubocop:
-        self.tempfile_suffix = 'rb'
-
-        path = self.filename
-        if not path:
-            # File is unsaved, and by default unsaved files use the default
-            # rubocop config because they do not technically belong to a folder
-            # that might contain a custom .rubocop.yml. This means the lint
-            # results may not match the rules for the currently open project.
-            #
-            # If the current window has open folders then we can use the
-            # first open folder as a best-guess for the current projects
-            # root folder - we can then pretend that this unsaved file is
-            # inside this root folder, and rubocop will pick up on any
-            # config file if it does exist:
-            folders = self.view.window().folders()
-            if folders:
-                path = os.path.join(folders[0], 'untitled.rb')
-
-        if path:
-            # With this path we can instead pass the file contents in via STDIN
-            # and then tell rubocop to use this path (to search for config
-            # files and to use for matching against configured paths - i.e. for
-            # inheritance, inclusions and exclusions).
-            #
-            # The 'force-exclusion' overrides rubocop's behavior of ignoring
-            # global excludes when the file path is explicitly provided:
-            command += ['--force-exclusion', '--stdin', path]
-            # Ensure the files contents are passed in via STDIN:
-            self.tempfile_suffix = None
-
-        return command
+    defaults = {
+        'selector': 'source.ruby - text.html - text.haml',
+        '--stdin:': '${file_path:$folder}'
```

Comment:
A heads-up, this doesn’t work, but instead needs to be:

```python
defaults = {
    'selector': 'source.ruby - text.html - text.haml',
    '--stdin:': '${file:$folder}'
```

The reason is that, per the [Sublime docs on Build Systems variables](https://www.sublimetext.com/docs/3/build_systems.html#variables), the `file_path` is

> The path to the folder that contains the file in the active view.

while `file` is

> The full path, including folder, to the file in the active view.

Using `file_path` means that Rubocop will lint that entire directory, which leads to some strange messages.

Otherwise, this is a great improvement!

Reply:
Actually, after digging in further in to bring these changes over to SublimeLinter-contrib-cookstyle, I’m not sure I see a way to deal with new, unsaved files this way, b/c of how `--stdin` treats a folder path (i.e., it lints everything in that path).

Reply:
We may have to lookup what rubocop actually does here, but it should still lint only what we pass in via `stdin` not the whole folder. Also: please post the strange messages of course.

Reply:
I misread the rubocop docs here, I thought that they want (now, t.i. in recent versions) a 'path' just to resolve the config files, so basically what otherwise the 'working_dir' usually is.

Reply:
Ahh, it’s possible that I’m using a version old enough that it expects something different.

Reply:
Ok, finally tried this out with the latest rubocop, and:

```bash
09:49:56 (0) rbyrnes at ATH036586 in ~/dev/chef-aws-jb/cookbooks/jb_base on master
♠ cat recipes/default.rb | rubocop -s recipes/default.rb
Inspecting 1 file
C

[snip]

1 file inspected, 6 offenses detected

09:50:30 (1) rbyrnes at ATH036586 in ~/dev/chef-aws-jb/cookbooks/jb_base on master
♠ cat recipes/default.rb | rubocop -s ./
Inspecting 20 files
CCCCCCCCCCCCCCCCCCCC

[snip]

20 files inspected, 128 offenses detected
```

So when the [Rubocop docs say](https://docs.rubocop.org/en/stable/basic_usage/#command-line-flags):

> Takes one argument, a path, relative to the root of the project.

They mean a relative path for the file you are piping in via STDIN. Passing a directory path instead of a path that either potentially or actually corresponds to what you’re piping will mean that every file in that directory is also linted, which is not what we want.

Reply:
Well, tell me that's not a bug in rubocop. 

We certainly don't pass in relative paths as well, and I actually looked at the source code of rubocop, I don't think that help message is true.

Reply:
I mean, sure feels buggy, or at least the docs are quite ambiguous. Probably worth engaging the rubocop folks to better understand the intent behind how that arg works.